lib/Sema/CSSimplify.cpp

```diff
@@ -13595,21 +13595,43 @@ ConstraintSystem::simplifyExplicitGenericArgumentsConstraint(
     }
 
     decl = overloadChoice.getDecl();
+
     auto openedOverloadTypes = getOpenedTypes(overloadLocator);
     openedTypes.append(openedOverloadTypes.begin(), openedOverloadTypes.end());
   }
 
-  auto genericContext = decl->getAsGenericContext();
-  if (!genericContext)
+  std::function<GenericParamList *(ValueDecl *)> getGenericParams =
+      [&](ValueDecl *decl) -> GenericParamList * {
+    auto genericContext = decl->getAsGenericContext();
+    if (!genericContext)
+      return nullptr;
+
+    auto genericParams = genericContext->getGenericParams();
+    if (!genericParams || genericParams->size() == 0) {
```

Comment:
I don't think you can have a non-null generic parameter list with zero parameters in it

Reply:
This was the original check but I can simplify it down to just !genericParams

Reply:
Done!